test/jdk/javax/xml/crypto/dsig/XalanTest.java

```diff
@@ -0,0 +1,135 @@
+ *          java.xml/com.sun.org.apache.xpath.internal.res
+ *          java.xml/com.sun.org.apache.xpath.internal
+ *          java.xml/com.sun.org.apache.xpath.internal.compiler
+ *
+ *
+ * @library /test/lib
+ * @compile -XDignore.symbol.file KeySelectors.java SignatureValidator.java
+ *     X509KeySelector.java ValidationTests.java
+ * @compile xalan/TransformXPath.java xalan/TransformXPath2Filter.java
+ * @run main/othervm XalanTest
+ * @author Sean Mullan
+ */
+import java.io.File;
+import java.io.FileInputStream;
+import java.security.*;
+import javax.xml.crypto.Data;
+import javax.xml.crypto.KeySelector;
+import javax.xml.crypto.OctetStreamData;
+import javax.xml.crypto.URIDereferencer;
+import javax.xml.crypto.URIReference;
+import javax.xml.crypto.URIReferenceException;
+import javax.xml.crypto.XMLCryptoContext;
+import javax.xml.crypto.dsig.XMLSignatureFactory;
+
+import jdk.test.lib.security.SecurityUtils;
+
+/**
+ * This test used to be part of ValidationTests but was moved into its own
+ * test because it tests a signature that contains the here() function which
+ * depends on Xalan internals. The Xalan dependency has been removed from
+ * the DSig implementation but can be optionally configured by using a
+ * customized TransformXPath instance. Use this test as an example.
+ */
+public class XalanTest {
+
+    private static SignatureValidator validator;
+    private final static String DIR = System.getProperty("test.src", ".");
+    private final static String DATA_DIR =
+        DIR + System.getProperty("file.separator") + "data";
+    private final static String XALAN_DIR =
+            DIR + System.getProperty("file.separator") + "xalan";
+    private final static String KEYSTORE =
+        DATA_DIR + System.getProperty("file.separator") + "certs" +
+        System.getProperty("file.separator") + "xmldsig.jks";
+    private final static String STYLESHEET =
+        "http://www.w3.org/TR/xml-stylesheet";
+    private final static String STYLESHEET_B64 =
+        "http://www.w3.org/Signature/2002/04/xml-stylesheet.b64";
+
+    public static void main(String args[]) throws Exception {
+        // Re-enable sha1 algs
+        SecurityUtils.removeAlgsFromDSigPolicy("sha1");
+        // Use custom XPath implementation
+        System.setProperty(
+                "com.sun.org.apache.xml.internal.security.resource.config",
+                "xalan/config-xalan.xml");
+
+        validator = new SignatureValidator(new File(XALAN_DIR));
+
+        KeyStore keystore = KeyStore.getInstance("JKS");
+        KeySelector ks;
+        try (FileInputStream fis = new FileInputStream(KEYSTORE)) {
+            keystore.load(fis, "changeit".toCharArray());
+            ks = new X509KeySelector(keystore, false);
+        }
+        if (!validator.validate("signature.xml", ks, new HttpURIDereferencer(), false)) {
+            throw new Exception
+                    ("At least one signature did not validate as expected");
```

Comment:
There's only one signature in this test now, so change message to "Signature did not validate as expected".